docs/userGuide/plugins/mermaid.md

```diff
@@ -0,0 +1,108 @@
+### Plugin: Mermaid
+
+<div id="content">
+
+This plugin allows you to utilize [Mermaid](https://mermaid-js.github.io/mermaid/) by automatically importing the library and initializing the rendering of the diagrams.
+
+> Mermaid is a JavaScript based diagramming and charting tool that renders Markdown-inspired text definitions to create and modify diagrams dynamically.
+
+<box type="info">
+
+All supported diagrams are available in [the Mermaid official documentation](https://mermaid-js.github.io/mermaid/).
+
+</box>
+
+To enable this plugin, add `mermaid` to your site's plugins.
+
+```js {heading="site.json"}
+{
+  ...
+  "plugins": [
+    "mermaid"
+  ]
+}
+```
+
+Optionally, you can specify an alternative URL to load the Mermaid code in the plugin context.
```

Comment:
Can we put this in a panel! I'm assuming most users would not need this functionality

Reply:
Also could we make it clearer what is `load the mermaid code`? 
Can include some explanation that the mermaid code is pulled from a CDN and hence this url specifies which is used

Reply:
Sure!